Felienne/hedy
# Make sleep command accept variables, expressions and list access

**Refactor the sleep command so that it accepts variables, arithmetic expressions and list access**

**Description**

- _Changes sleep command's grammar to support multiple arguments: variables, expressions and list access_
- _Adds type validation, so that the sleep commands accepts only integers_
- _Adds runtime type validation, so that sleep arguments are transformed to a number, e.g. when used with ask command_
- _Automatically updates the sleep argument to fit the range [0, 10], to avoid public programs causing program hanging_

**Fixes #1753 **

**How to test**

1) Consider what would be the acceptable maximum limit for the sleep argument. We do not want users to supply huge values (e.g. 9999999), because this surely will yield an error. The current max value is 10, which means 10 seconds. Is this value too low?

Run Hedy locally and go to level 4.
2) Execute the following snippet and confirm that the error you receive guides you correctly what to correct.
```
n is "test"
sleep n
```
3) Execute the following snippet and supply a non-numerical value. Check that the error makes sense. Note that this error is available only in English atm.
```
n is ask "How long?"
sleep n
```
